### PR TITLE
Added ZoomToBounds components

### PR DESCRIPTION
**Pull request by egabrielse**
**Main Features:**
1. Added and tested two components to manage automatic zoom to the boundaries of a county or to the state:
    - ZoomToBoundsButton
    - ZoomToBoundsMenu (sets up a vertical list of ZoomToBoundsButton components) 
 
2. ZoomToBoundsMenu was added to App.js along with two supporting functions. 

3 Using the Bootstrap module, restructured App.js so that ZoomToBoundsMenu displays alongside the map.

**Added Modules:**
- @turf/bbox (https://www.npmjs.com/package/@turf/bbox) 
- @testing-library/user-event  (https://testing-library.com/docs/ecosystem-user-event/)
- react-bootstrap / bootstrap (https://react-bootstrap.github.io/)

**TODO (in a seperate PR):**
- Styling of components to match the UX mock (https://xd.adobe.com/embed/5431252b-dd5c-4140-91d9-83c5fa4f8691-0162/)

**Gif Examples:**
zoom_to_cook_county

zoom_back_to_state

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -106,11 +107,23 @@ export default class App extends Component {
    */
   renderTooltip() {
     const {hoveredCounty, hoveredZipCode, x, y} = this.state;
+    const style = {
```
Comment (mcdevittbass) [inline src/App.js:110]: Is there a reason you don't want to give the components classNames and put the styles in a CSS file? Just thinking about cleaning it up a little.

File: src/App.js [hunk 2]
```diff
@@ -121,36 +134,84 @@ export default class App extends Component {
...
+      }
+    });
+  }
+
+  /**
+   * Returns a list of the county GeoJSON features
+   */
+  getIllinoisCountyFeatures = () => {
+    if (this.state.illinois_counties != null) {
+      var countyFeatures =  this.state.illinois_counties.features.map((feature) => {
```
Comment (mcdevittbass) [inline src/App.js:156]: //just condensing this to one line!
const countyFeatures =  this.state.illinois_counties.features.map((feature) => feature)

File: src/App.js [hunk 3]
```diff
@@ -121,36 +134,84 @@ export default class App extends Component {
...
+        return feature;
+      })
+      const sortedCountyFeatures = countyFeatures.sort((a,b) => (a.properties.NAME > b.properties.NAME) ? 1 : -1);
+
+      return sortedCountyFeatures;
+    } else {
```
Comment (mcdevittbass) [inline src/App.js:162]: else is not necessary here - again just a cosmetic comment

File: src/App.js (hunk 1 above)
Comment (egabrielse) [reply src/App.js:110]: There actually are two named components that replace that: ZipCodeSnapshotOverlay and CountySnapshotOverlay. They are implemented by Map.js, which will take the mapping logic out of App.js. I had left App.js pretty much alone for the time being, until we had sorted out the state-management situation. I think a following ticket will be to refactor App.js

Comment (mcdevittbass) [reply src/App.js:110]: Ah okay, thanks!

File: src/App.js (hunk 2 above)
Comment (egabrielse) [reply src/App.js:156]: Changed

File: src/App.js (hunk 3 above)
Comment (egabrielse) [reply src/App.js:162]: Changed

**Review by mcdevittbass — APPROVED** [review]
Looks like good steps!

**Review by ellenglight — APPROVED** [review]
Approving this on behalf of @mcdevittbass since she doesn't have write permissions yet

**Pull request merged**